mountpoint-s3/src/data_cache/in_memory_data_cache.rs

```diff
@@ -42,12 +42,17 @@ impl<Key: Eq + Hash> DataCache<Key> for InMemoryDataCache<Key> {
         self.block_size
     }
 
-    fn cached_block_indices(&self, cache_key: &Key, range: Range<BlockIndex>) -> DataCacheResult<Vec<BlockIndex>> {
+    fn cached_block_indices<R: RangeBounds<BlockIndex>>(
+        &self,
+        cache_key: &Key,
+        range: R,
+    ) -> DataCacheResult<Vec<BlockIndex>> {
         let data = self.data.read().unwrap();
-        let result = match data.get(cache_key) {
+        let mut result = match data.get(cache_key) {
             None => Vec::new(),
-            Some(blocks) => range.into_iter().filter(|idx| blocks.contains_key(idx)).collect(),
+            Some(blocks) => blocks.keys().filter(|idx| range.contains(idx)).copied().collect(),
```

Comment:
Doesn't this break the guarantee about the order of the blocks?
```
/// Indices in the vector are already sorted.
```

Reply:
Vector is sorted on line 55. It does mean we need to explicitly sort, but this implementation is only for testing.